installer.sh

```diff
@@ -3,8 +3,19 @@
 # NOTE: Heavily inspired by get-stack.hs script for installing stack.
 # https://raw.githubusercontent.com/commercialhaskell/stack/stable/etc/scripts/get-stack.sh
 
-HOME_LOCAL_BIN="$HOME/.local/bin"
-HOME_LOCAL_SHARE="$HOME/.local/share"
+# Default install directory
+INSTALL_DIR="$HOME"
+
+while getopts ":d:" opt; do
+  case $opt in
+    d) INSTALL_DIR="$OPTARG" ;;
+    *) echo "Invalid argument: -$OPTARG" >&2; exit 1 ;;
+  esac
+done
+
+# Assign the directories 
+HOME_LOCAL_BIN="$INSTALL_DIR/.local/bin"
+HOME_LOCAL_SHARE="$INSTALL_DIR/.local/share"
```

Comment:
Good idea with the installation destination!
How come you decided to do this, I'm guessing you wanted a global installation option?

Unfortunately, these locations (i.e., `.local/something`) are not arbitrary  - the way we organize our files isn't independent of the top-level installation destination. In other words, we should probably avoid replicating their structure without considering where we're putting them.

Coincidentally, this is exactly what @infomiho and me have been talking about for the last couple of days! Check this comment for the full explanation (it's a whole thing with a long history): https://github.com/wasp-lang/wasp/pull/953#discussion_r1071544116

Anyway:
 - If you just wanted the option to install Wasp globally, the XDG standard tells us exactly what where we should put our files - we can simply add the `-g/--global` flag instead of this one and "hardcode" those paths.
 - If you wanted to give users full control over where Wasp is installed, we'll discuss how to best approach it.

Reply:
This was intended to make working with online IDEs (Replit, GitPod) a bit easier, as we sometimes need to change the installation destination to get it working correctly on these environements. See: https://github.com/wasp-lang/replit-template/tree/vince-add-replit-template

Reply:
Yup, this was per my suggestion, and I didn't think much about uninstalling at that point!

I would suggest the following, to keep it simple: we go with this, but don't tell users about it, so it is mostly just for us and potentially for some power users.

The first version of uninstall, the simple one, will work for everyone that use default paths. It won't work for us and maybe for some very rare power users, but in that case we assume us/them know what they are doing anyway, for now.

Once we do a solution with XDG paths, we can refactor all of this, but then we can do it in one coherent go. I wouldn't let that stop us right now though to get out what we need. But would certainly, in an issue for XDG paths, mention that right now we have an option to install in different than default path, that we need it for replit, and that we should make sure we can still do what we need for replit.